test/Npgsql.Tests/BatchTests.cs

```diff
@@ -169,6 +215,38 @@ public async Task NpgsqlBatchCommand_StatementType()
             Assert.That(batch.BatchCommands[6].StatementType, Is.EqualTo(StatementType.Other));
         }
 
+        [Test]
+        public async Task NpgsqlBatchCommand_StatementType15()
```

Comment:
nit: maybe just add a single MERGE case to NpgsqlBatchCommand_StatementType just above? All the other statement types are already tested this way.

Reply:
Combined those 2 with some if blocks